native/src/utils/DatabaseConnector.ts

```diff
@@ -695,14 +693,19 @@ class DatabaseConnector {
     return BlobUtil.fs.exists(path)
   }
 
-  async readFile(path: string): Promise<string> {
+  async readFile<T>(path: string): Promise<T> {
```

Comment:
What would happen if the Types do not match. 
E.g. `readFile<CityResourceCacheJsonType>(path)` does not return `CityResourceCacheJsonType`

Is this catched somewhere? Should it?

Reply:
No, it is not caught anywhere. I am not sure if that is even possible to assert that. If so, it should be done in a separate issue though.

Reply:
you could assert that if we had more information on the path. But not sure wether this is worth the work